androidApp/src/main/java/com/mbta/tid/mbta_app/android/routeDetails/CollapsableStopList.kt

```diff
@@ -0,0 +1,157 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.fromHtml
+import androidx.compose.ui.unit.dp
+import com.mbta.tid.mbta_app.android.R
+import com.mbta.tid.mbta_app.android.component.HaloSeparator
+import com.mbta.tid.mbta_app.android.component.StopListRow
+import com.mbta.tid.mbta_app.android.component.StopRowStyle
+import com.mbta.tid.mbta_app.android.stopDetails.TripRouteAccents
+import com.mbta.tid.mbta_app.android.util.Typography
+import com.mbta.tid.mbta_app.model.RouteCardData
+import com.mbta.tid.mbta_app.model.RouteDetailsStopList
+
+@Composable
+fun CollapsableStopList(
+    lineOrRoute: RouteCardData.LineOrRoute,
+    segment: RouteDetailsStopList.Segment,
+    onClick: (RouteDetailsStopList.Entry) -> Unit,
+    isLastSegment: Boolean = false,
+    rightSideContent: @Composable RowScope.(RouteDetailsStopList.Entry, Modifier) -> Unit,
+) {
+
+    var stopsExpanded by rememberSaveable { mutableStateOf(false) }
+
+    if (segment.stops.size == 1) {
+        val stop = segment.stops.first()
+        StopListRow(
+            stop.stop,
+            onClick = { onClick(stop) },
+            routeAccents = TripRouteAccents(lineOrRoute.sortRoute),
+            modifier =
+                Modifier.minimumInteractiveComponentSize().background(colorResource(R.color.fill1)),
+            connectingRoutes = stop.connectingRoutes,
+            stopRowStyle = StopRowStyle.StandaloneStop,
+            descriptor = {
+                Text(stringResource(R.string.less_common_stop), style = Typography.footnote)
+            },
+            rightSideContent = { modifier -> rightSideContent(stop, modifier) },
+        )
+    } else {
+        Column {
+            Column(Modifier.padding(horizontal = 6.dp)) {
+                Row(
+                    Modifier.height(IntrinsicSize.Min)
+                        // TODO: Click label
+                        .clickable() { stopsExpanded = !stopsExpanded }
+                        // TODO: Content description
+                        .padding(horizontal = 10.dp)
+                        .defaultMinSize(minHeight = 48.dp),
+                    verticalAlignment = Alignment.CenterVertically,
+                ) {
+                    AnimatedContent(
+                        stopsExpanded,
+                        transitionSpec = {
+                            fadeIn(animationSpec = tween(500)) togetherWith
+                                fadeOut(animationSpec = tween(500))
+                        },
+                    ) {
+                        Row(
+                            verticalAlignment = Alignment.CenterVertically,
+                            horizontalArrangement = Arrangement.Center,
+                            modifier = Modifier.padding(start = 22.dp).width(20.dp),
+                        ) {
+                            if (it) {
```

Comment:
I find it confusing that this `it` is the lambda argument from the `AnimatedContent` rather than the `Row`; naming that argument would make it clearer.